lib/Page.js

```diff
@@ -71,6 +80,37 @@ Page.prototype.collectIncludedFiles = function (dependencies) {
   });
 };
 
+/**
+ * Records the front matter data of the current page into this.frontMatter
+ * @param includedPage a page with its dependencies included
+ */
+Page.prototype.collectFrontMatter = function (includedPage) {
+  return new Promise((resolve) => {
+    const $ = cheerio.load(includedPage);
+    const frontMatter = $('frontmatter');
+    const includedFrontMatter = $('frontmatter div');
+    if (frontMatter.get(FIRST_ELEMENT_INDEX) !== undefined) {
+      let frontMatterData = '';
+      if (includedFrontMatter.get(FIRST_ELEMENT_INDEX) !== undefined) {
```

Comment:
I think you can use `.first()` instead of `get(FIRST_ELEMENT_INDEX)`.
Also, `children().first()`

Also, is there only one/do we only care about one `includedFrontMatter`?

Reply:
For both cases of frontmatter, only the attributes from the first occurrence of `<frontmatter>` tag is taken to represent the data of that particular page. 

`frontmatter` is for front matter that is defined in the page itself, whereas `includedFrontMatter` is for front matter that has been included from another file. It is possible that there may be multiple occurrences of the `<frontmatter>` tag in a single page.